# Generated by Django 4.1.3 on 2025-03-13 17:12

import CalorieCounter.accounts.validators
import CalorieCounter.fitness.validators
import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Trainer',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('first_name', models.CharField(error_messages={'max_length': 'Maximum name length is 50 characters!'}, max_length=50, validators=[CalorieCounter.fitness.validators.validate_name])),
                ('last_name', models.CharField(error_messages={'max_length': 'Maximum name length is 50 characters!'}, max_length=50, validators=[CalorieCounter.fitness.validators.validate_name])),
                ('gender', models.CharField(choices=[('Male', 'Male'), ('Female', 'Female')], max_length=6)),
                ('birthday', models.DateField(validators=[CalorieCounter.accounts.validators.validate_age_through_birthday])),
                ('biography', models.TextField(blank=True, error_messages={'max_length': 'Maximum length of your biography is 250 characters!'}, max_length=250, null=True)),
                ('profile_pic_url', models.URLField(error_messages={'url': 'Invalid URL!'}, validators=[django.core.validators.URLValidator()])),
            ],
            options={
                'ordering': ('first_name', 'last_name'),
            },
        ),
    ]
